subsys/net/lib/nrf_cloud/src/nrf_cloud_credentials.c

```diff
@@ -147,6 +147,29 @@ int nrf_cloud_credentials_configured_check(void)
 		}
 	}
 
+	if (IS_ENABLED(CONFIG_NRF_CLOUD_MQTT) || IS_ENABLED(CONFIG_NRF_CLOUD_REST)) {
+		if (!cs.ca_combined && !cs.ca_aws && cs.ca_coap) {
+			/* There is a CA, but not large enough to be correct */
+			LOG_ERR("Cannot connect using MQTT or REST; "
+			        "AWS root CA certificate is missing but CoAP root CA is present.");
+			ret = -ENOTSUP;
```

Comment:
how about a different return code to indicate that credentials were found, but the size isn't correct for the configured transport type?